### DISCUSSION TITLE
Google Cloud unexpected compute engine costs using xDrip+/Nightscout

### DISCUSSION THREAD

**Discussion by emp-00** [NightscoutFoundation/xDrip#3802] [Q&A]
Dear Team, I've been using Google Cloud with xDrip+ [using navid200's instructions](https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html) (already running since 2 years). Costs have been 1 or 2 €-cents each month. Just recently I received a budget alert from Google and checking my cost management report I can see that in the first 7 days of December I have already 7 €-cents of costs for "Compute Engine". 

This is strange - I have not installed anything new on my NS instance and there's nothing special ongoing, just my xDrip+ CGM values pushed to NS and data pulled from NS via webbrowser access. Nothing unusual, doing the same as always... Checking CPU and RAM usage of the VM for the last 14 days - I can only see constant load, cannot identify where those costs are coming from. From the report I conclude those costs are due to usage on one single day: December 6th (yesterday)? Will this continue or is this just a "one day anomaly"?

Does anybody have a clue for this? Thanks for tipps.

![image](https://github.com/user-attachments/assets/0a1a5b9d-d92b-42ac-9d3f-5eb7279d37a5)

![image](https://github.com/user-attachments/assets/0b7db5d5-68dd-42f5-a7b1-da780dde592c)

**Comment by Navid200** [comment]
I have no way of predicting what will happen assuming something has changed.

I don't think there is anything new about this.
In order to qualify for free tier, you have to satisfy several conditions.  Some of those conditions are about what happens over a month.  The month of December is not over yet.  Therefore, there is no way to know if you will satisfy the free tier conditions by the end of the month.
My understanding is that what you are seeing reflects the cost without the free tier discounts accounted for.

That's the same reason when you create a new virtual machine, you will see a monthly estimate cost of $7.31 as shown on this page: https://navid200.github.io/xDrip/docs/Nightscout/VirtualMachine.html  

You say you received a budget alert.  I am not sure what you mean by that.

**Reply by emp-00** [reply]
I have not changed anything with the Google Cloud or Nightscout setup - same usage since 2 years. That's why I'm surprised and a bit worried about this development. Look at the screenshots, I have no idea where those "Compute Engine" costs are suddenly coming from.

The budget alert came via Email (I have configured it to 5% of budget, never received such an alert before):
![image](https://github.com/user-attachments/assets/bcb8736c-57c1-48c4-a898-c01e186f35ee)

**Comment by gigich** [comment]
Hi
I have the same problem and it is because of outgoing network traffic. See also this topic:
[https://github.com/NightscoutFoundation/xDrip/discussions/3848](url)
Lukas
